src/Compilers/VisualBasic/Portable/Symbols/NamedTypeSymbol.vb

```diff
@@ -1089,6 +1099,85 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+
+        Friend Overridable ReadOnly Property HasRequiredMembersError As Boolean
+            Get
+                TryCalculateRequiredMembersWithCache(Nothing)
+                Debug.Assert(Not _lazyRequiredMembers.IsDefault)
+                Return _lazyRequiredMembers = s_requiredMembersErrorSentinel
+            End Get
+        End Property
+
+        Friend MustOverride ReadOnly Property HasAnyDeclaredRequiredMembers As Boolean
+
+        Private Function TryCalculateRequiredMembersWithCache(ByRef requiredMembersBuilder As ImmutableSegmentedDictionary(Of String, Symbol).Builder) As Boolean
+            Debug.Assert(requiredMembersBuilder Is Nothing)
+            Dim lazyRequiredMembers As ImmutableSegmentedDictionary(Of String, Symbol) = _lazyRequiredMembers
+
+            If Not lazyRequiredMembers.IsDefault Then
+                If lazyRequiredMembers = s_requiredMembersErrorSentinel Then
+                    Return False
+                Else
+                    requiredMembersBuilder = lazyRequiredMembers.ToBuilder()
+                    Return True
+                End If
+            End If
+
+            Dim success = TryCalculateRequiredMembers(requiredMembersBuilder)
+
+            Dim requiredMembers = If(success,
+                If(requiredMembersBuilder?.ToImmutable(), ImmutableSegmentedDictionary(Of String, Symbol).Empty),
+                s_requiredMembersErrorSentinel)
+
+            RoslynImmutableInterlocked.InterlockedInitialize(_lazyRequiredMembers, requiredMembers)
+
+            Return success
+        End Function
+
+        Private Function TryCalculateRequiredMembers(ByRef requiredMembersBuilder As ImmutableSegmentedDictionary(Of String, Symbol).Builder) As Boolean
+            If BaseTypeNoUseSiteDiagnostics?.TryCalculateRequiredMembersWithCache(requiredMembersBuilder) = False Then
+                Return False
+            End If
+
+            If (Not HasAnyDeclaredRequiredMembers) AndAlso requiredMembersBuilder Is Nothing Then
+                Return True
+            End If
+
+            If requiredMembersBuilder Is Nothing Then
+                requiredMembersBuilder = ImmutableSegmentedDictionary.CreateBuilder(Of String, Symbol)()
+            End If
+
+            For Each member In GetMembersUnordered()
+                Dim existingMember As Symbol = Nothing
+                ' Need to make sure that members from a base type weren't hidden by members from the current type. That is an error scenario
+                If requiredMembersBuilder.TryGetValue(member.Name, existingMember) Then
+                    ' This is only permitted if the member is an override of a required member from a base type, and is required itself
+                    Dim overriddenMember = TryCast(member, PropertySymbol)?.OverriddenProperty
+
+                    If (Not member.IsRequired()) OrElse
+                       overriddenMember Is Nothing OrElse
+                       overriddenMember IsNot existingMember Then
```

Comment:
>overriddenMember IsNot existingMember [](http://example.com/codeflow?start=23&length=37)

This might fail to match members in a generic context for example.

Reply:
Looks like this is correct, added `GenericSubstitution_InheritanceAndOverride_NoneSet`